pkg/engine/deployment.go

```diff
@@ -220,6 +220,9 @@ func newDeployment(ctx *Context, info *deploymentContext, opts deploymentOptions
 				if imp.PluginDownloadURL == "" {
 					imp.PluginDownloadURL = defaultProviderInfo[imp.Type.Package()].PluginDownloadURL
 				}
+				if imp.PluginChecksums != nil {
```

Comment:
Do you mean here if `imp.PluginChecksums == nil` instead? Maybe also good to check that `defaultProviderInfo` actually has that package checksum:
```go
checksums, ok := defaultProviderInfo[imp.Type.Package()]
if ok {
  imp.PluginChecksums = checksums
}

```